gnomad_qc/v5/sample_qc/create_sample_qc_metadata_ht.py

```diff
@@ -0,0 +1,634 @@
+    finalized_outlier_filtering,
+    get_gen_anc_ht,
+    hard_filtered_samples,
+    related_samples_to_drop,
+    relatedness,
+)
+from gnomad_qc.v4.resources.meta import meta as v4_meta
+from gnomad_qc.v4.resources.variant_qc import TRUTH_SAMPLES
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("sample_metadata")
+logger.setLevel(logging.INFO)
+
+
+def restructure_meta_fields(meta_ht: hl.Table) -> hl.Table:
+    """
+    Restructure the meta Table to group related fields into nested structs.
+
+    .. note::
+
+        Creates:
+        - `sex_imputation`: struct with sex karyotype and mean depth
+        - `project_meta`: struct with project-level metadata
+        - `metrics`: struct with sequencing and QC metrics
+
+    :param meta_ht: Table with metadata.
+    :return: Annotated Table with new structured fields.
+    """
+    meta_ht = meta_ht.transmute(
+        sex_imputation=hl.struct(
+            sex_karyotype=meta_ht.sex_karyotype,
+            mean_dp=meta_ht.mean_depth,
+        )
+    )
+
+    meta_ht = meta_ht.transmute(
+        project_meta=hl.struct(
+            age=meta_ht.age,
+            releasable=meta_ht.releasable,
+            sample_source=meta_ht.sample_source,
+            data_type=meta_ht.data_type,
+            project=meta_ht.project,
+        ),
+        metrics=hl.struct(
+            contamination_freemix=meta_ht.contamination_freemix,
+            contamination_charr=meta_ht.contamination_charr,
+            chimeras_rate=meta_ht.chimeras_rate,
+            bases_over_20x_coverage=meta_ht.bases_over_20x_coverage,
+            median_insert_size=meta_ht.median_insert_size,
+            callrate=meta_ht.callrate,
+        ),
+    )
+
+    return meta_ht
+
+
+def prepare_meta_with_hard_filters(
+    meta_ht: hl.Table,
+    samples_to_exclude: hl.expr.SetExpression,
+    aou_hard_filters_ht: hl.Table,
+) -> hl.Table:
+    """
+    Add hard filter annotations to the metadata Table
+
+    :param meta_ht: Table with metadata.
+    :param samples_to_exclude: Table with samples to exclude.
+    :return: Annotated meta Table with hard filter fields
+    """
+    # Build exclusion list.
+    exclusion_ht = hl.Table.parallelize(
+        [{"s": x} for x in hl.eval(samples_to_exclude)]
+    ).key_by("s")
+
+    exclusion_ht = add_project_prefix_to_sample_collisions(
+        t=exclusion_ht, sample_collisions=sample_id_collisions.ht(), project="aou"
+    )
+
+    samples_to_exclude = hl.literal(exclusion_ht.s.collect())
+
+    # Obtain AoU hard filters and set hard_filter to sample_qc_metrics if any hard filters were applied.
+    aou_ht = meta_ht.filter(meta_ht.project == "aou").select()
+
+    aou_ht = aou_ht.annotate(hard_filters=aou_hard_filters_ht[aou_ht.key])
+
+    aou_ht = aou_ht.annotate(
+        hard_filters=hl.if_else(
+            hl.is_defined(aou_ht.hard_filters) & (hl.len(aou_ht.hard_filters) > 0),
+            {"sample_qc_metrics"},
```

Comment:
Lets put the full set of hard filters in the annotation
```suggestion
            aou_ht.sample_qc_metric_hard_filters,
```